adapters/adapterstest/test_json.go

```diff
@@ -331,3 +317,84 @@ func diffJson(t *testing.T, description string, actual []byte, expected []byte)
 		}
 	}
 }
+
+// testMakeRequestsImpl asserts the resulting values of the bidder's `MakeRequests()` implementation
+// against the expected JSON-defined results and makes sure we do not incurr in data races in the
+// process. To verify no data races occur this function creates:
+//  1) A shallow copy of the unmarshalled openrtb2.BidRequest that will provide reference values to
+//     shared memory that we don't want the adapters' implementation of `MakeRequests()` to modify.
+//  2) A deep copy that will preserve the original values of all the fields. This copy remains untouched
+//     by the adapters' processes and serves as reference of what the shared memory values should still
+//     be after the `MakeRequests()` call.
+//
+// The original values stored in the deep copy will be compared against the shared memory values and
+// discrepancies will point to data race conditions in an adapter's `MakeRequests()` implementation.
+//
+// Because neither the shallow nor the deep copies are passed to `MakeRequests()` we know static fields
+// in openrtb2.BidRequest nor openrtb2.Imp elements will change, therefore, we can reliably compare
+// reference values using the `assert.Equal()` method.
```

Comment:
Didn't end up liking this huge comment. It probably needs rewording. Suggestions?